components/base/sensorcontrolled/spin.go

```diff
@@ -5,97 +5,44 @@ import (
 	"math"
 	"time"
 
-	"github.com/golang/geo/r3"
-	"github.com/pkg/errors"
-	"go.viam.com/utils"
-
-	"go.viam.com/rdk/components/movementsensor"
 	rdkutils "go.viam.com/rdk/utils"
 )
 
 const (
-	increment = 0.01 // angle fraction multiplier to check
-	oneTurn   = 360.0
+	increment   = 0.01 // angle fraction multiplier to check
+	oneTurn     = 360.0
+	slowDownAng = 15. // angle from goal for spin to begin breaking
 )
 
-// Spin commands a base to turn about its center at a angular speed and for a specific angle.
+// Spin commands a base to turn about its center at an angular speed and for a specific angle.
 func (sb *sensorBase) Spin(ctx context.Context, angleDeg, degsPerSec float64, extra map[string]interface{}) error {
```

Comment:
This function is doing a lot! would recommend that you think about breaking it up. Doesn't have to be in this PR.

Reply:
kk, tried to break up the function abit but couldn't get it into a form that I liked, since most of the steps interact with the request parameters in some way the functions were ending up pretty ugly, and they wouldn't really be reusable anywhere